### DISCUSSION TITLE
Gledopto Downlight leave&join network

### DISCUSSION THREAD

**Discussion by JLSP** [Koenkk/zigbee2mqtt#12594] [Devices]
Hi all, I am getting an odd issue in some of my GL-D-004P devices. I have paired successfully twelve GL-D-004P devices in my zigbee2mqtt network but one of them started to leave and join the network every minute, I have been checking the downlight but no clues about the issue.

Today I have found the same issue in another device, it has started to leave and join the network every minute. It is happening only in this type of device.

This is the state of a member in the same group:

{
    "brightness": 254,
    "color": {
        "x": 0.414,
        "y": 0.3949
    },
    "color_mode": "color_temp",
    "color_temp": 298,
    "linkquality": 76,
    "state": "OFF",
    "update": {
        "state": "available"
    },
    "update_available": true
}

The state of the reconnecting device when it is available is:

{
    "color_mode": "xy",
    "linkquality": 61,
    "state": "ON",
    "update": {
        "state": null
    },
    "update_available": null,
    "brightness": 254,
    "color": {
        "h": 121,
        "hue": 121,
        "s": 99,
        "saturation": 99,
        "x": 0.1778,
        "y": 0.712
    },
    "color_temp": 158
}

This is the full conversation of the device:

 2022-05-25 17:47:57Failed to read state of 'PB_Cocina_DL4' after reconnect (Cannot read property 'defaultSendRequestWhen' of undefined)
Error 2022-05-25 17:48:01Failed to read state of 'PB_Cocina_DL4' after reconnect (Cannot read property 'defaultSendRequestWhen' of undefined)
Info 2022-05-25 17:48:02Device 'PB_Cocina_DL4' joined
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"PB_Cocina_DL4","ieee_address":"0xa4c138d0cdd085c7"},"type":"device_joined"}'
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"PB_Cocina_DL4"},"type":"device_connected"}'
Info 2022-05-25 17:48:02Starting interview of 'PB_Cocina_DL4'
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"PB_Cocina_DL4","ieee_address":"0xa4c138d0cdd085c7","status":"started"},"type":"device_interview"}'
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"PB_Cocina_DL4"},"type":"pairing"}'
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"PB_Cocina_DL4","ieee_address":"0xa4c138d0cdd085c7"},"type":"device_announce"}'
Info 2022-05-25 17:48:02MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"PB_Cocina_DL4"},"type":"device_announced"}'
Info 2022-05-25 17:48:02Successfully interviewed 'PB_Cocina_DL4', device has successfully been paired
Info 2022-05-25 17:48:02Device 'PB_Cocina_DL4' is supported, identified as: Gledopto Zigbee 9W Downlight RGB+CCT (pro) (GL-D-004P)
Info 2022-05-25 17:48:03MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Zigbee 9W Downlight RGB+CCT (pro)","exposes":[{"features":[{"access":7,"description":"On/off state of this light","name":"state","property":"state","type":"binary","value_off":"OFF","value_on":"ON","value_toggle":"TOGGLE"},{"access":7,"description":"Brightness of this light","name":"brightness","property":"brightness","type":"numeric","value_max":254,"value_min":0},{"access":7,"description":"Color temperature of this light","name":"color_temp","presets":[{"description":"Coolest temperature supported","name":"coolest","value":150},{"description":"Cool temperature (250 mireds / 4000 Kelvin)","name":"cool","value":250},{"description":"Neutral temperature (370 mireds / 2700 Kelvin)","name":"neutral","value":370},{"description":"Warm temperature (454 mireds / 2200 Kelvin)","name":"warm","value":454},{"description":"Warmest temperature supported","name":"warmest","value":500}],"property":"color_temp","type":"numeric","unit":"mired","value_max":500,"value_min":150},{"access":7,"description":"Color temperature after cold power on of this light","name":"color_temp_startup","presets":[{"description":"Coolest temperature supported","name":"coolest","value":150},{"description":"Cool temperature (250 mireds / 4000 Kelvin)","name":"cool","value":250},{"description":"Neutral temperature (370 mireds / 2700 Kelvin)","name":"neutral","value":370},{"description":"Warm temperature (454 mireds / 2200 Kelvin)","name":"warm","value":454},{"description":"Warmest temperature supported","name":"warmest","value":500},{"description":"Restore previous color_temp on cold power on","name":"previous","value":65535}],"property":"color_temp_startup","type":"numeric","unit":"mired","value_max":500,"value_min":150},{"description":"Color of this light in the CIE 1931 color space (x/y)","features":[{"access":7,"name":"x","property":"x","type":"numeric"},{"access":7,"name":"y","property":"y","type":"numeric"}],"name":"color_xy","property":"color","type":"composite"},{"description":"Color of this light expressed as hue/saturation","features":[{"access":7,"name":"hue","property":"hue","type":"numeric"},{"access":7,"name":"saturation","property":"saturation","type":"numeric"}],"name":"color_hs","property":"color","type":"composite"}],"type":"light"},{"access":2,"description":"Triggers an effect on the light (e.g. make light blink for a few seconds)","name":"effect","property":"effect","type":"enum","values":["blink","breathe","okay","channel_change","finish_effect","stop_effect"]},{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"GL-D-004P","options":[{"access":2,"description":"Controls the transition time (in seconds) of on/off, brightness, color temperature (if applicable) and color (if applicable) changes. Defaults to `0` (no transition).","name":"transition","property":"transition","type":"numeric","value_min":0},{"access":2,"description":"When enabled colors will be synced, e.g. if the light supports both color x/y and color temperature a conversion from color x/y to color temperature will be done when setting the x/y color (default true).","name":"color_sync","property":"color_sync","type":"binary","value_off":false,"value_on":true}],"supports_ota":true,"vendor":"Gledopto"},"friendly_name":"PB_Cocina_DL4","ieee_address":"0xa4c138d0cdd085c7","status":"successful","supported":true},"type":"device_interview"}'
Info 2022-05-25 17:48:03Configuring 'PB_Cocina_DL4'
Info 2022-05-25 17:48:03MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Zigbee 9W Downlight RGB+CCT (pro)","friendly_name":"PB_Cocina_DL4","model":"GL-D-004P","supported":true,"vendor":"Gledopto"},"type":"pairing"}'
Info 2022-05-25 17:48:03MQTT publish: topic 'zigbee2mqtt/PB_Cocina_DL4', payload '{"color_mode":"xy","linkquality":65,"state":null,"update":{"state":null},"update_available":null}'
Info 2022-05-25 17:48:03MQTT publish: topic 'zigbee2mqtt/PB_Cocina_DL4', payload '{"color_mode":"xy","linkquality":69,"state":null,"update":{"state":null},"update_available":null}'
Info 2022-05-25 17:48:03Successfully configured 'PB_Cocina_DL4'
Info 2022-05-25 17:48:04MQTT publish: topic 'zigbee2mqtt/PB_Cocina_DL4', payload '{"color_mode":"xy","linkquality":69,"state":"ON","update":{"state":null},"update_available":null}'
Warning 2022-05-25 17:48:23Device 'PB_Cocina_DL4' left the network
Info 2022-05-25 17:48:23MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"PB_Cocina_DL4","ieee_address":"0xa4c138d0cdd085c7"},"type":"device_leave"}'
Info 2022-05-25 17:48:23MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"left_network","meta":{"friendly_name":"0xa4c138d0cdd085c7"},"type":"device_removed"}'

Any idea about why this devices have this behaviour?

**Comment by Sunxperous** [comment]
I believe I am encountering the same issue with three of my GL-D-005P (12W) lights. After trying to reset (perhaps to no avail), now one of them is stuck in RGB mode, while the other two keep rejoining. The two lights also visibly turn off for 1 second, and then on again at the cool white temperature (which I believe was the last setting before this problem started).

Have you managed to find a solution?

**Reply by JLSP** [reply]
My problem was in the firmware version, so to update it I had to change my coordinator to an old CC2530. I reconnected all the downlights to the new coordinator and updated the firmware to version 10551203 via OTA, when they are updated I switch again to my real coordinator. I hope it helps for you.

**Reply by Sunxperous** [reply]
Thanks, I managed to perform a firmware update with my CC2652 for some reason that I described in https://github.com/Koenkk/zigbee2mqtt/issues/11202#issuecomment-1173047823.

**Comment by Sunxperous** [comment]
For others who chance upon this thread: this is covered in https://github.com/Koenkk/zigbee2mqtt/issues/11202 with solutions available.
